build.js

```diff
@@ -28,49 +30,60 @@ function unrollEnum(col) {
 
 var checks = [
 
-	function intCheck(column) {
+	function intCheck(intf, column) {
+    
 		var checks = ''
-		if (maxIntValues[column.dataType]) {
-			checks += '.number().integer()'
+        var bytes = intf.intToBytes[column.dataType];
+
+		if (bytes) {
 
-			var min = 0
-			var max = maxIntValues[column.dataType]
-			if (column.columnType.indexOf('unsigned') === -1) {
-				max = getSignedValue(max)
-				min = -1 * (max + 1)
+			checks += '.number().integer()'
+            
+			if (!intf.hasUnsigned) { 
+			    let [min, max] = bytesToSignedRange(bytes);
+                checks += '.max(' + max + ').min(' + min + ')'
 			}
+            else if(column.columnType.indexOf('unsigned') !== -1) {
```

Comment:
It looks like both of these `if` blocks contain the same code, so this should probably be `if (!intf.hasUnsigned || column.columnType.indexOf('unsigned') !== -1)`